From 9152f87b593cc51d3e56dddd2e93d1101065d8cc Mon Sep 17 00:00:00 2001
From: Florian Westphal <fwestpha@redhat.com>
Date: Tue, 2 Sep 2014 10:14:33 -0400
Subject: [net] ipv6: Need to sock_put on csum error

Message-id: <1409652920-431-11-git-send-email-fwestpha@redhat.com>
Patchwork-id: 90169
O-Subject: [PATCH 10/57] [RHEL7.1 net] ipv6: Need to sock_put on csum error
Bugzilla: 1131999
RH-Acked-by: Jiri Pirko <jpirko@redhat.com>
RH-Acked-by: Vlad Yasevich <vyasevic@redhat.com>
RH-Acked-by: Nikolay Aleksandrov <nikolay@redhat.com>
RH-Acked-by: Ivan Vecera <ivecera@redhat.com>
RH-Acked-by: Jiri Benc <jbenc@redhat.com>

BZ: 1131999

commit 79e0f1c9f2c7096e93879d4158f2d34142b693dd
Author: Tom Herbert <therbert@google.com>
Date:   Mon May 5 16:43:58 2014 -0700

    ipv6: Need to sock_put on csum error

    Commit 4068579e1e098fa81d48db9ba4432ab664c58561 ("net: Implmement
    RFC 6936 (zero RX csums for UDP/IPv6)") introduced zero checksums
    being allowed for IPv6, but in the case that a socket disallows a
    zero checksum on RX we need to sock_put.

    Signed-off-by: Tom Herbert <therbert@google.com>
    Signed-off-by: David S. Miller <davem@davemloft.net>

Signed-off-by: Jiri Benc <jbenc@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/net/ipv6/udp.c b/net/ipv6/udp.c
index 9c2f5f4..89b5bea 100644
--- a/net/ipv6/udp.c
+++ b/net/ipv6/udp.c
@@ -854,6 +854,7 @@ int __udp6_lib_rcv(struct sk_buff *skb, struct udp_table *udptable,
   int ret;
 
   if (!uh->check && !sk->sk_no_check) {
+   sock_put(sk);
    udp6_csum_zero_error(skb);
    goto csum_error;
   }
-- 
1.7.1